tests/bluetooth/hci_uart_async/src/test_hci_uart_async.c

```diff
@@ -70,10 +70,10 @@ static void hci_uart_thread_entry(void *p1, void *p2, void *p3)
 }
 static int sys_init_spawn_hci_uart(void)
 {
-	k_thread_name_set(&hci_uart_thread, "hci_uart_main");
 	k_thread_create(&hci_uart_thread, hci_uart_thread_stack,
 			K_THREAD_STACK_SIZEOF(hci_uart_thread_stack), hci_uart_thread_entry, NULL,
 			NULL, NULL, CONFIG_MAIN_THREAD_PRIORITY, 0, K_NO_WAIT);
+	k_thread_name_set(&hci_uart_thread, "hci_uart_main");
```

Comment:
Not a hard nack, but it used to work to set the name before starting the thread. This has the benefit of the thread always having the correct name from the very start. I suspect it should still work if we don't call `arch_thread_name_set` if the thread is not running yet, and it will be called later from `z_setup_new_thread`.

Reply:
I agree. This change is not really needed after I've added
```c
	thread_status = thread->callee_saved.thread_status;
	if (!thread_status) {
		return -EAGAIN;
	}
```
Now we always set thread name after creation (checked with spatch)

Reply:
Wasn't this code anyhow incorrect (irrespective of other changes in this PR)?
`k_thread_create()` clears the stored thread name

Reply:
> Wasn't this code anyhow originally incorrect (irrespective of other changes in this PR)? `k_thread_create()` clears the stored thread name

It looks like you are right. Never mind then.